pepdbagent/modules/annotation.py

```diff
@@ -0,0 +1,341 @@
+        :param limit: return limit
+        :param offset: return offset
+        :return: pydantic model: AnnotationReturnModel(
+            limit:
+            offset:
+            count:
+            result: List [ AnnotationModel(
+                        namespace:
+                        name:
+                        tag:
+                        is_private:
+                        number_of_samples:
+                        description:
+                        last_update_date:
+                        submission_date:
+                        digest:
+                    )
+                ]
+            )
+        """
+        if all([namespace, name, tag]):
+            found_annotations = list(
+                self._get_single_annotation(
+                    namespace=namespace,
+                    name=name,
+                    tag=tag,
+                    admin=admin,
+                )
+            )
+            return AnnotationReturnModel(
+                count=len(found_annotations),
+                limit=1,
+                offset=1,
+                result=found_annotations,
+            )
+        return AnnotationReturnModel(
+            limit=limit,
+            offset=offset,
+            count=self._count_projects(
+                namespace=namespace, search_str=query, admin=admin
+            ),
+            result=self._get_projects(
+                namespace=namespace,
+                search_str=query,
+                admin=admin,
+                offset=offset,
+                limit=limit,
+            ),
+        )
+
+    def get_by_rp(
+        self,
+        registry_paths: Union[List[str], str],
+        admin: Union[List[str], str] = None,
+    ) -> AnnotationReturnModel:
+        """
+        Get project annotations by providing registry_path or list of registry paths.
+        :param registry_paths: registry path string or list of registry paths
+        :param admin: list of namespaces where user is admin
+                :return: pydantic model: AnnotationReturnModel(
+            limit:
+            offset:
+            count:
+            result: List [ AnnotationModel(
+                        namespace:
+                        name:
+                        tag:
+                        is_private:
+                        number_of_samples:
+                        description:
+                        last_update_date:
+                        submission_date:
+                        digest:
+                    )
+                ]
+            )
+        """
+        if isinstance(registry_paths, list):
+            anno_results = []
+            for path in registry_paths:
+                try:
+                    namespace, name, tag = registry_path_converter(path)
+                except RegistryPathError as err:
+                    _LOGGER.error(str(err), registry_paths)
+                    continue
```

Comment:
So, this comes up a lot in pephub, and this is a good example. Sometimes when we submit requests to pephub, it appears that they were successful when they were not since `pepdbagent` catches errors and just continues.

One example is project submission or project finding. If you go to pephub and try to submit a project with the same name, `pepdbagent` will catch the error (I believe) and pephub just keeps going without knowing what happened.

So, I think we should throw Exceptions when users make a mistake like supplying an incorrect registry path, if that makes sense. Otherwise it can lead to people thinking something was successful when it was not.

Reply:
Throwing an error is the easiest solution and I would add it everywhere, but I don't think we should do it.
why? -> Because we don't want to close connection. I know that connection is reopening every time when we call some function, but I would avoid closing connection.
I think solution is: standardize error message ¯\_(ツ)_/¯

Reply:
I agree with Nathan. pepdbagent should just be throwing errors. It's at the PEPhub level that they should be caught.,

If pephub catches the error, it doesn't mean the connection will be closed.

Reply:
@Khoroshevskyi I think you shouldn't think of pepdbagent as the final consumer, who must keep running in case of errors. You should think of it as a low-level connection. Throw lots of specific types of Errors that can be caught and then handled by whatever the calling software is.

PEPhub is the ultimate consumer, so all the catching of the specific errors should be handled there.

Reply:
If pepdbagent isn't final consumer it makes perfect sense. Now I understand how it should work. I will add Exceptions to pepdbagent